avocado/utils/asset.py

```diff
@@ -684,9 +685,15 @@ def remove_asset_by_path(cls, asset_path):
 
         :param asset_path: full path of the asset file.
         """
-        os.remove(asset_path)
-        filename = f"{asset_path}-CHECKSUM"
-        os.remove(filename)
```

Comment:
I would keep the `os.remove`. I don't see any additional benefit with usage of `pathlib`, especially when it is inconsistent with other parts of asset utility which uses  `os.remove`.